### DISCUSSION TITLE
Adding a new Object Store

### DISCUSSION THREAD

**Discussion by thhapke** [apache/arrow-rs#5393] [General]
I have developed a RustAPI to  an yet unsupported object store (SAP HDLFS). Now I like to use the object store API as a wrapper to further enable arrow-rs of using HDLFS. To be honest I am a bit lost how to start best. I am quite new to rust and it would be great if there is a kind of guide which I could use to put my builder and api-calls into it. Many thanks.

**Comment by tustvold** [comment]
The ObjectStore trait can be implemented in third-party crates which would be my recommendation for this. 

https://github.com/datafusion-contrib/datafusion-objectstore-hdfs might be a useful starting point

**Reply by thhapke** [reply]
Many thanks. I have started to implement the object store trait to my API using your link as a guide. I also found some good ideas from the AWS implementation.
